src/microbe_stage/editor/EditorActionHistory.cs

```diff
@@ -191,6 +169,37 @@ public bool HexPlacedThisSession<THex>(THex hex)
         return History.OfType<HexPlacementActionData<THex>>().Any(a => a.PlacedHex == hex);
     }
 
+    /// <summary>
+    ///   Pick the data with which you can regain maximum MP to combine.
+    /// </summary>
+    /// <param name="currentData">The data you want to combine</param>
+    /// <param name="previousData">A list of data currentData may combine with</param>
+    /// <returns>
+    ///   Three-element tuple:
+    ///   CostDelta is non-positive, indicating the amount of MP you will regain;
+    ///   MinimumCostActionData is the action data to combine with (not yet combined);
+    ///   Mode is the interference mode currentData has with MinimumCostActionData.
+    /// </returns>
+    private static (int CostDelta, EditorCombinableActionData MinimumCostActionData, ActionInterferenceMode Mode)
+        GetLeastCostCombineTuple(EditorCombinableActionData currentData,
```

Comment:
```suggestion
        GetLeastCostCombinedTuple(EditorCombinableActionData currentData,
```

Reply:
The action data returned is not combined. I think renaming like that will make confusions.

Reply:
Well the original name confused me already... Maybe I can come up with a better suggestion if you briefly explain what this method actually does? I don't think I'm yet mentally prepared to the brain melting trying to read and understand this PR in its entirety is going to cause...

Reply:
Added in the description